### ISSUE TITLE
Rule "Boolean literals should not be redundant" do not handle nullable booleans

### ISSUE THREAD

**Issue by DuAell** [SonarSource/SonarTS#606]
**SonarQube version:** Version 7.0 (build 36138)<br>

**Rule key:** S1125 

**Reproducer** 
```
Sockets.All(x => x.IsStateValid == true); // False-positive here
```

**Expected behavior**
bool? should not be handled as bool, and no bugs should be detected here as IsStateValid could be null

**Comment by stas-vilchik** [comment]
@DuAell could you tell me how the type of `IsStateValid` is defined, and if you're using `strict` or `strictNullChecks` compiler options?

**Comment by DuAell** [comment]
Oh yes sorry, IsStateValid is a nullable boolean

public bool? IsStateValid { get; set; }

**Comment by DuAell** [comment]
Sorry, bad repo, error was for C#
